File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeValueMvoStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class AttributeValueMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'attributeValueIdAttributeId' => 'string',
            'attributeValueIdValue' => 'string',
            'attributeVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeValueMvoStateEventId
     */
    private $value;

    /**
     * @param AttributeValueMvoStateEventId $value
     */
    public function __construct(AttributeValueMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeValueMvoStateEventId();
        }
    }

    /**
     * @return AttributeValueMvoStateEventId
     */
    public function toAttributeValueMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getAttributeValueIdAttributeId()
    {
        return $this->value->getAttributeValueId()->getAttributeId();
    }

    /**
     * @param string $attributeValueIdAttributeId
     */
    public function setAttributeValueIdAttributeId($attributeValueIdAttributeId)
    {
        $this->value->getAttributeValueId()->setAttributeId($attributeValueIdAttributeId);
    }

    /**
     * @return string
     */
    public function getAttributeValueIdValue()
    {
        return $this->value->getAttributeValueId()->getValue();
    }

    /**
     * @param string $attributeValueIdValue
     */
    public function setAttributeValueIdValue($attributeValueIdValue)
    {
        $this->value->getAttributeValueId()->setValue($attributeValueIdValue);
    }

    /**
     * @return Long
     */
    public function getAttributeVersion()
    {
        return $this->value->getAttributeVersion();
    }

    /**
     * @param Long $attributeVersion
     */
    public function setAttributeVersion($attributeVersion)
    {
        $this->value->setAttributeVersion($attributeVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getAttributeValueIdAttributeId(),
            $this->getAttributeValueIdValue(),
            $this->getAttributeVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return AttributeValueMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setAttributeValueIdAttributeId($pValues[0]);
        $this->setAttributeValueIdValue($pValues[1]);
        $this->setAttributeVersion($pValues[2]);
        return $this;
    }

}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AttributeValueMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeValueMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "attributeValueIdAttributeId",
            "attributeValueIdValue",
            "attributeVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeValueMvoStateEventId value;

    public AttributeValueMvoStateEventIdFlattenedDto()
    {
        this(new AttributeValueMvoStateEventId());
    }

    public AttributeValueMvoStateEventIdFlattenedDto(AttributeValueMvoStateEventId value)
    {
        this.value = value;
    }

    public AttributeValueMvoStateEventId toAttributeValueMvoStateEventId()
    {
        return this.value;
    }


    public String getAttributeValueIdAttributeId()
    {
        return this.value.getAttributeValueId().getAttributeId();
    }

    public void setAttributeValueIdAttributeId(String attributeValueIdAttributeId)
    {
        this.value.getAttributeValueId().setAttributeId(attributeValueIdAttributeId);
    }

    public String getAttributeValueIdValue()
    {
        return this.value.getAttributeValueId().getValue();
    }

    public void setAttributeValueIdValue(String attributeValueIdValue)
    {
        this.value.getAttributeValueId().setValue(attributeValueIdValue);
    }

    public Long getAttributeVersion()
    {
        return this.value.getAttributeVersion();
    }

    public void setAttributeVersion(Long attributeVersion)
    {
        this.value.setAttributeVersion(attributeVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeValueMvoStateEventIdFlattenedDto other = (AttributeValueMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

